javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java

```diff
@@ -47,6 +47,11 @@
   private static ClassLoader AGENT_CLASSLOADER = null;
 
   public static void initialize(Instrumentation inst, URL bootstrapUrl) {
+    if (AGENT_CLASSLOADER != null) {
+      // don't run initialization multiple times, initialize can be called multiple times via dynamic attach
```

Comment:
Just confirming, do we need the field to be volatile if accessed multiple times? With dynamic attach, is `initialize` guaranteed to read the memory from a previous update?

Reply:
I would guess we get no such guarantee, it's only accessed a few times, so little harm making it volatile, good call 👍

Reply:
Alternatively if we cared about affecting the other reads of the variable on first init, we could add an `AtomicBoolean` only for checking reinitialization.